.cursor/rules/sitecore.mdc

```diff
@@ -0,0 +1,146 @@
+---
+description: Sitecore SDK-specific rules for XM Cloud development
+alwaysApply: false
+globs:
+  - '**/*sitecore*.ts'
+  - '**/*sitecore*.tsx'
+  - '**/*content*.ts'
+  - '**/*content*.tsx'
+  - '**/sitecore.config.ts'
+  - '**/sitecore.cli.config.ts'
+  - 'src/api/**'
+  - 'src/lib/**'
+---
+
+# Sitecore SDK Rules
+
+## XM Cloud Integration
+
+Configuration:
+
+- Always use environment variables for API endpoints and keys
+- Never hardcode API keys in source code
+- Use `.env.example` files to document required environment variables
+- Prefer `sitecore.config.ts` for centralized configuration
+
+```typescript
+// sitecore.config.ts
+export const sitecoreConfig = {
+  sitecoreApiHost: process.env.SITECORE_API_HOST || '',
+  sitecoreApiKey: process.env.SITECORE_API_KEY || '',
+  siteName: process.env.SITECORE_SITE_NAME || 'default',
+};
+```
+
+API Client Usage:
+
+- Use HTTPS for all XM Cloud endpoints
+- Implement proper retry logic with exponential backoff
+- Cache responses appropriately (consider content freshness)
+- Handle GraphQL errors gracefully
+
+## Component Development
+
+Sitecore Component Naming:
+
+- Use descriptive, feature-based names: `HeroWithContent`, `ProductListing`, `ContentBlockGrid`
+- Follow PascalCase convention
+- Include component type in name when helpful: `LayoutContainer`, `ContentRenderer`
+
+Component Registration:
+
+- Register components in the component map
+- Use consistent component names across registration and files
+- Include TypeScript types for all component props
+
+```typescript
+// Component props interface
+interface ContentBlockProps {
+  fields: {
+    title: Field;
+    content: Field;
+    image: Field;
+  };
+}
+```
+
+## Content Management
+
+Field Handling:
+
+- Always validate field existence before rendering
+- Use helper functions for common field operations
+- Handle empty/null fields gracefully
+- Prefer Sitecore field components over manual rendering
+
+```typescript
+// Good: Using Sitecore field components
+<Text field={fields?.title} tag="h1" />
+<RichText field={fields?.content} />
+
+// Avoid: Manual field value extraction unless necessary
+```
+
+Layout Service:
+
+- Use the Layout Service for page data fetching
```

Comment:
Perhaps we should instruct it to prefer using our scClient methods, with potential for expansion